src/main.zig

```diff
@@ -109,7 +109,9 @@ pub fn serialize(comptime T: type, allocator: Allocator, data: T, list: *ArrayLi
                 .Slice => {
                     // Simple case: string
                     if (@sizeOf(ptr.child) == 1) {
-                        try list.append(128 + @as(u8, @truncate(data.len)));
+                        if (data.len != 1) {
+                            try list.append(128 + @as(u8, @truncate(data.len)));
+                        }
```

Comment:
For 1 byte slices, we must encode it as a single byte.
For other cases (length > 1, **and length 0**), we encode it as a string.